cmd/cluster/health.go

```diff
@@ -96,59 +98,120 @@ func (o *healthOptions) run() error {
 		healthObject.Expected.Worker = int(cluster.Nodes().Compute())
 	}
 
-	awsClient, err := osdCloud.GenerateAWSClientForCluster(o.awsProfile, o.clusterID)
-	if err != nil {
-		return err
-	}
-
-	instances, err := awsClient.DescribeInstances(&ec2.DescribeInstancesInput{})
-	if err != nil {
-		return err
-	}
 	runningMasters := 0
 	runningInfra := 0
 	runningWorkers := 0
 	totalStopped := 0
 	totalCluster := 0
 
-	//Here we count the number of customer's running worker, infra and master instances in the cluster in the given region. To decide if the instance belongs to the cluster we are checking the Name Tag on the instance.
-	for idx := range instances.Reservations {
-		for _, inst := range instances.Reservations[idx].Instances {
-			tags := inst.Tags
-			for _, t := range tags {
-				if *t.Key == "Name" {
-					if strings.HasPrefix(*t.Value, cluster.InfraID()) && strings.Contains(*t.Value, "master") {
-						totalCluster += 1
-						if *inst.State.Name == "running" {
-							runningMasters += 1
-						}
-						if *inst.State.Name == "stopped" {
-							totalStopped += 1
-						}
-
-					} else if strings.HasPrefix(*t.Value, cluster.InfraID()) && strings.Contains(*t.Value, "infra") {
-						totalCluster += 1
-						if *inst.State.Name == "running" {
-							runningInfra += 1
-						}
-						if *inst.State.Name == "stopped" {
-							totalStopped += 1
-						}
-					} else if strings.HasPrefix(*t.Value, cluster.InfraID()) && strings.Contains(*t.Value, "worker") {
-						totalCluster += 1
-						if *inst.State.Name == "running" {
-							runningWorkers += 1
-						}
-						if *inst.State.Name == "stopped" {
-							totalStopped += 1
-						}
-
+	if cluster.CloudProvider().ID() == "gcp" {
+		clusterResources, err := ocmClient.ClustersMgmt().V1().Clusters().Cluster(o.clusterID).Resources().Live().Get().Send()
+		if err != nil {
+			return err
+		}
+		projectClaimRaw := clusterResources.Body().Resources()["gcp_project_claim"]
```

Comment:
is "gcp_project_claim" guaranteed to be present here? I'm wondering if it could be helpful to add a check here, like

```go
projectClaimRaw, ok := clusterResources.Body().Resources()["gcp_project_claim"]
if !ok { ... }
```
```